### PR TITLE
Refactor SQL query for enhanced security

### PR DESCRIPTION
**Pull request by vegimcarkaxhija**
- Replaced direct string concatenation with DbQuery class in the query.
- Added input sanitization using pSQL to prevent SQL injection vulnerabilities.

The update ensures that the query handling adheres to PrestaShop's best practices, enhancing both code maintainability and security.

Also fixed Idin, after testing the new changes I saw that idin was not appearing at all in checkout

### CODE REVIEW

File: src/Form/Modifier/ProductFormModifier.php [hunk 1]
```diff
@@ -75,7 +75,12 @@ public function modify(
      */
     private function getBuckarooIdinForProduct(int $productId): ?bool
     {
-        $sql = 'SELECT buckaroo_idin FROM ' . _DB_PREFIX_ . 'bk_product_idin WHERE product_id = ' . (int) $productId;
+        $sql = new \DbQuery();
+
+        $sql->select('buckaroo_idin');
+        $sql->from('bk_product_idin');
+        $sql->where('product_id = ' . pSQL($productId));
```
Comment (harli91) [inline src/Form/Modifier/ProductFormModifier.php:82]: why aren't you using prepared statements?

**Review by harli91 — CHANGES_REQUESTED** [review]

File: src/Form/Modifier/ProductFormModifier.php (hunk 1 above)
Comment (vegimcarkaxhija) [reply src/Form/Modifier/ProductFormModifier.php:82]: Method 'setParameter' not found in \DbQuery

Comment (ShuCh3n) [reply src/Form/Modifier/ProductFormModifier.php:82]: I dont see setParameter.

https://devdocs.prestashop-project.org/8/development/components/database/dbquery/

**Review by ShuCh3n — APPROVED** [review]

File: src/Form/Modifier/ProductFormModifier.php (hunk 1 above)
Comment (harli91) [reply src/Form/Modifier/ProductFormModifier.php:82]: sorry, saw this code https://github.com/buckaroo-it/PrestaShop/blob/develop/src/Repository/PaymentMethodRepository.php#L43-L44, but from what I see this uses the entity manager query builder not the \DbQuery class

**Pull request merged**